### PR TITLE
fix: Fix NOL pay bug - fetching of the cards

### PR DESCRIPTION
**Pull request by borisprimer**
Make sure fetching of the card is being called after fetching of the secret key is finished

# Description

[CHKT-2121](https://primerapi.atlassian.net/browse/CHKT-2121)

- Fix a bug in the case where fetching of the cards was called right after init() of the NolPayLinkedCardsComponent.
Fetching of the card was called before fetching the nolSdk secret key from the backend, I added these two requests into the DispatchGroup() 

There were no changes in the public nol API and no changes in the documentation are needed.

# Other Notes

- Other changes that are not specifically related to the intent of the PR

# Manual Testing

_Add manual testing notes here if applicable, otherwise remove this section_

# Screenshots

_If applicable, otherwise remove this section_

# Contributor Checklist

- [x]  All status checks have passed prior to code review
- [ ]  I have correctly prefixed one of the following conventional commit titles:
    - [ ]  chore - no version update
    - [x]  fix - patch version update
    - [ ]  feat - minor version update
    - [ ]  BREAKING CHANGE - major version update
- [ ]  I have added unit tests to a reasonable level of coverage where suitable
- [ ]  I have added UI tests to new user flows, if applicable
- [x]  I have manually tested newly added UX
- [ ]  I have open a documentation PR, if applicable

# Reviewer Checklist

- [ ]  I have verified that a suitable set of automated tests has been added
- [ ]  I have verified that the title prefix aligns to the code changes + whether a release is expected after merging the PR
- [ ]  I have verified the documentation PR aligns with this PR, if applicable

# Before Merging

- [x]  If introducing a breaking change, I have communicated it internally
- [x]  Any related documentation PRs are ready to merge

# Other Stuff

- You can find out more about our automation checks [here](https://primerio.notion.site/iOS-Automation-Checks-198a1eb0e8994d999fb696d5902d97bb)
- Find out more about conventional commits [here](https://primerio.notion.site/Conventional-Commits-6ecfd6a0269a4db2af76d9f0537936b3)

[CHKT-2121]: https://primerapi.atlassian.net/browse/CHKT-2121?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Sources/PrimerSDK/Classes/Core/PrimerHeadlessUniversalCheckout/Composable/NolPay/NolPayLinkedCardsComponent.swift [hunk 1]
```diff
@@ -19,19 +19,45 @@ public class NolPayLinkedCardsComponent: PrimerHeadlessComponent {
     var countryCode: String?
     var phoneMetadataService: NolPayPhoneMetadataProviding?
 
-    public init() {
```
[the remark this answers is not in the document]
Comment (NQuinn27) [reply Sources/PrimerSDK/Classes/Core/PrimerHeadlessUniversalCheckout/Composable/NolPay/NolPayLinkedCardsComponent.swift:22]: Good shout - mitigating factor here could be that there is just one merchant using this right now, correct?  
SDK v3 for this is not appealing.

Questions:
1. Did Android have the same bug, how did they tackle it?
2. Is there a way to fix this bug without changing the public interface? 
2a. If yes and we choose this one, is this change to the DX a big enough benefit to make a breaking change for.
2b. Linking in with point 1, does this change keep us in line with Android DX?

Friendly reminder: don't forget to update the docs when this goes live :)

[the remark this answers is not in the document]
Comment (borisprimer) [reply Sources/PrimerSDK/Classes/Core/PrimerHeadlessUniversalCheckout/Composable/NolPay/NolPayLinkedCardsComponent.swift:22]: 1. I removed init so it falls back to the default init
2. I suggest jumping on a branch opening the whole file and checking it, GitHub code review is confusing to me sometimes. API is still the same as before, `public func getLinkedCardsFor(mobileNumber: String, completion: @escaping (Result<[PrimerNolPaymentCard], PrimerError>) -> Void)` is still the only public function besides init()

[the remark this answers is not in the document]
Comment (jnewc) [reply Sources/PrimerSDK/Classes/Core/PrimerHeadlessUniversalCheckout/Composable/NolPay/NolPayLinkedCardsComponent.swift:22]: Ok, I've looked at both files side by side now and it makes sense 👍 

I think the most surprising thing is that (TIL) the auto-synthesized init of a public class is public, not internal - but if it works, it works 🤷

File: Sources/PrimerSDK/Classes/Core/PrimerHeadlessUniversalCheckout/Composable/NolPay/NolPayLinkedCardsComponent.swift [hunk 2]
```diff
@@ -13,25 +13,53 @@ public class NolPayLinkedCardsComponent: PrimerHeadlessComponent {
...
-                                            ],
-                                            diagnosticsId: UUID().uuidString)
+                                                 userInfo: ["file": #file,
+                                                            "class": "\(Self.self)",
+                                                            "function": #function,
+                                                            "line": "\(#line)"],
+                                                 diagnosticsId: UUID().uuidString)
             ErrorHandler.handle(error: error)
             self.errorDelegate?.didReceiveError(error: error)
             return
```
Comment (jnewc) [inline Sources/PrimerSDK/Classes/Core/PrimerHeadlessUniversalCheckout/Composable/NolPay/NolPayLinkedCardsComponent.swift:65]: Not 100% sure on this but will suggest anyway: should `nolPay` be set to `nil` in each of these guards?

**Review by jnewc — APPROVED** [review]

File: Sources/PrimerSDK/Classes/Core/PrimerHeadlessUniversalCheckout/Composable/NolPay/NolPayLinkedCardsComponent.swift (hunk 2 above)
Comment (borisprimer) [reply Sources/PrimerSDK/Classes/Core/PrimerHeadlessUniversalCheckout/Composable/NolPay/NolPayLinkedCardsComponent.swift:65]: Not sure what am I getting with this?

**Pull request merged**